pkg/provision/primitives/counters.go

```diff
@@ -184,6 +186,35 @@ type resourceUnits struct {
 	CRU uint64 `json:"cru,omitempty"`
 }
 
+// CheckMemoryRequirements checks memory requirements for a reservation and compares it to whats in the counters
+// and what is the total memory on this node
+func (c *Counters) CheckMemoryRequirements(r *provision.Reservation, memStats *mem.VirtualMemoryStat) error {
```

Comment:
This method doesn't use anything from the `Counters` object. So we could just extract it from there.

Reply:
Yes it does use the `c.CurrentUnits()`, also this function uses the private functions in order to calculate the requested HRU by a workload. notice `processVolume` and `processKubernetes`

Reply:
It was placed here to avoid import cycles